mlflow/xgboost.py

```diff
@@ -340,6 +341,9 @@ def record_eval_results(eval_results):
             """
 
             def callback(env):
+                batch_metrics_handler.record_metrics(
```

Comment:
Can we add `batch_metrics_handler` as an argument to `record_eval_results` and thread it through to this `callback` to ensure that we're not accidentally referencing some state left over from a previous `BatchMetricsHandler`, for example?